src/lib.rs

```diff
@@ -101,20 +93,89 @@ impl<T> LocatedSpan<T> {
     /// # fn main() {
     /// let span = LocatedSpan::new(b"foobar");
     ///
-    /// assert_eq!(span.offset,     0);
-    /// assert_eq!(span.line,       1);
-    /// assert_eq!(span.column,     0);
-    /// assert_eq!(span.fragment,   &b"foobar"[..]);
+    /// assert_eq!(span.offset,         0);
+    /// assert_eq!(span.line,           1);
+    /// assert_eq!(span.get_column(),   1);
+    /// assert_eq!(span.fragment,       &b"foobar"[..]);
     /// # }
     /// ```
     pub fn new(program: T) -> LocatedSpan<T> {
          LocatedSpan {
              line: 1,
-             column: 0,
              offset: 0,
              fragment: program
          }
     }
+
+    fn get_columns_and_bytes_before(&self) -> (usize, &[u8]) {
+        let self_bytes = self.fragment.as_bytes();
+        let self_ptr = self_bytes.as_ptr();
+        let before_self = unsafe {
+            let orig_input_ptr = self_ptr.offset(-(self.offset as isize));
```

Comment:
probably wiser to add this right before: `assert!(self.offset < isize::max_value());`